projects/ngx-translate-router/src/lib/localize-router.service.ts

```diff
@@ -178,19 +185,31 @@ export class LocalizeRouterService {
       const previousLang = this.parser.getLocationLang(previousEvent.url) || this.parser.defaultLang;
       const currentLang = this.parser.getLocationLang(currentEvent.url) || this.parser.defaultLang;
 
-      if (currentLang !== previousLang) {
-        // mutate router config directly to avoid getting out of sync
-        this.parser.mutateRouterRootRoute(currentLang, previousLang, this.router.config);
+      if (currentLang !== previousLang && this.latestUrl !== currentEvent.url) {
+        this.latestUrl = currentEvent.url;
```

Comment:
you can remove this line while using the same line on 201.

Reply:
If you test, you will see that the line 201 will only be called after the second navigation end.
So we can't remove this line. Line 201 is only not have an "else".